### ISSUE TITLE
cmd/compile: internal compiler error: bad type: chan struct { v [65536]byte }

### ISSUE THREAD

**Issue by burik666** [golang/go#49767]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.17.2 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.3"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/app/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2607095789=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
**main.go**
```go
package main

func main() {
    ch := make(chan struct{ v [65536]byte })
    close(ch)
}
```
`go build `
### What did you expect to see?
```
# t
./main.go:4:13: channel element type too large (>64kB)
```

### What did you see instead?
```
# t
./main.go:4:12: internal compiler error: bad type: chan struct { v [65536]byte }
```

**Comment by heschi** [comment]
cc @danscales @randall77 @mdempsky

**Comment by griesemer** [comment]
See #49771 for the possible remedies.

**Comment by danscales** [comment]
One part of my change  https://golang.org/cl/367955 could be backported to 1.17 to fix this issue in 1.17.  But the issue doesn't seem serious enough to require a backport.  Let me know if anyone disagrees.

**Comment by griesemer** [comment]
@danscales Is there anything left to do with this issue?

**Comment by danscales** [comment]
No, I don't think we need to backport the fix to 1.17 (since even though it is an internal compiler error, it gives a pretty good indication of what the problem is anyway).  So, I'll close this.  People can feel free to re-open if they really think we should backport to 1.17.
